### ISSUE TITLE
linter: `react/react-compiler` generates false “Todo: Unsupported declaration type for hoisting” on generic components/hooks

### ISSUE THREAD

**Issue by acusti** [oxc-project/oxc#23611]
### What version of Oxlint are you using?

1.70.0

### What command did you run?

`oxlint -c .oxlintrc.json Select.tsx`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["react"],
  "categories": { "correctness": "off" },
  "rules": {
    // reportAllBailouts surfaces the compiler’s silent "Todo" bailouts, i use it as a compiler-compatibility gate
    "react/react-compiler": ["error", { "reportAllBailouts": true }]
  }
}
```

### What happened?

`react/react-compiler` with `reportAllBailouts: true` reports a false `Todo: Unsupported declaration type for hoisting` bailout on components/hooks that use type generics. the same code compiles successfully under the React team’s `babel-plugin-react-compiler@1.0.0`, so this is a false positive specific to the oxc integration.

**Minimal reproduction:**

```ts Select.tsx
export function Select<T extends string>({
  options,
  onChange,
}: {
  options: T[];
  onChange: (value: T) => void;
}) {
  const handleChange = (value: T) => {
    onChange(value);
  };

  return (
    <select onChange={(event) => handleChange(event.target.value as T)}>
      {options.map((option) => (
        <option key={option} value={option}>
          {option}
        </option>
      ))}
    </select>
  );
}
```

**Actual:**
```
Select.tsx:1:8: error react(react-compiler): Todo: Unsupported declaration type for hoisting
  help: variable "T" declared with Unknown
```

**Expected:** no error (the component compiles and memoizes fine under `babel-plugin-react-compiler`, verified `CompileSuccess`).

**Trigger:** a type parameter referenced in a type annotation inside a nested function (here the `handleChange` arrow’s `(value: T)` / `as T`). removing `<T extends string>` clears the error.

**Likely root cause (FWIW):**

1. in `crates/oxc_react_compiler/src/convert_scope.rs`, the first pass (`Create all bindings from symbols`) iterates every `scoping.symbol_ids()` with no value/type filter, so TS type parameters become value bindings; `get_binding_kind` has no type-only arm, so a type parameter falls through to `BindingKind::Unknown` / `declaration_type: "Unknown"`
2. a type-position reference to it inside a nested function then makes the upstream HIR hoisting pass emit this `Todo`

a guard like `if flags.is_type() && !flags.is_value() { continue; }` (skipping the symbol and its references) when building `ScopeInfo` would exclude type-only symbols. the SWC bridge may want the same check.

**Comment by acusti** [comment]
@Boshen heads up that i posted a PR as #23613 to address this if it’s helpful, but all good if not

**Comment by acusti** [comment]
@Boshen i just upgraded to `oxlint@1.74.0`, and i was able to remove all my oxlint-disable comments related to this bug. i took a quick look at recent commits in the react-compiler crate and seem like the fix landed in 854ef8d. thanks for doing that! i’ll close this out now.
